src/components/split-button/split-button.component.js

```diff
@@ -17,108 +17,99 @@ const marginPropTypes = filterStyledSystemMarginProps(
 
 const CONTENT_WIDTH_RATIO = 0.75;
 
-class SplitButton extends Component {
-  constructor(props) {
-    super(props);
-    this.buttonLabelId = guid();
-    this.additionalButtons = [];
-    this.listening = false;
-    this.isToggleButtonFocused = false;
-    this.userInputType =
-      "ontouchstart" in document.documentElement ? "touchstart" : "click";
-
-    this.splitButtonNode = React.createRef();
-    this.buttonContainer = React.createRef();
-  }
-
-  state = {
-    showAdditionalButtons: false,
-    minWidth: 0,
-  };
-
-  focusToggleButton = () => {
-    this.isToggleButtonFocused = true;
-    this.showButtons();
-  };
-
-  showButtons = () => {
-    document.addEventListener(this.userInputType, this.handleClickOutside);
-    this.setState({
-      showAdditionalButtons: true,
-      minWidth:
-        CONTENT_WIDTH_RATIO *
-        this.splitButtonNode.current.getBoundingClientRect().width,
-    });
-
-    if (!this.listening) {
-      document.addEventListener("keydown", this.handleKeyDown);
-      this.listening = true;
-    }
-  };
-
-  hideButtons = () => {
-    if (this.isToggleButtonFocused) return;
+const SplitButton = ({
+  align = "left",
+  as = "secondary",
+  buttonType,
+  children,
+  disabled = false,
+  iconPosition = "before",
+  iconType,
+  onClick,
+  size = "medium",
+  subtext,
+  text,
+  ...rest
+}) => {
+  const isToggleButtonFocused = useRef(false);
+  const listening = useRef(false);
+  const buttonLabelId = useRef(guid());
+  const additionalButtons = useRef([]);
+  const splitButtonNode = useRef();
+  const buttonContainer = useRef();
+  const [showAdditionalButtons, setShowAdditionalButtons] = useState(false);
+  const [minWidth, setMinWidth] = useState(0);
+  const userInputType = useRef(
+    "ontouchstart" in document.documentElement ? "touchstart" : "click"
+  );
+
+  const hideButtons = useCallback(() => {
+    if (isToggleButtonFocused.current) return;
+
+    setShowAdditionalButtons(false);
+    listening.current = false;
+  }, []);
+
+  const handleClickOutside = useCallback(
+    ({ target }) => {
+      if (
+        !splitButtonNode.current.contains(target) &&
+        buttonContainer.current &&
+        !buttonContainer.current.contains(target)
+      ) {
+        hideButtons();
+      }
+    },
+    [hideButtons]
+  );
+
+  const handleKeyDown = useCallback(
+    (ev) => {
+      if (!listening.current) {
```

Comment:
Kinda confusing because to be precise the listeners are still attached. Maybe it would be better to use the `showAdditionalButtons` here?

Reply:
this is a leftover variable from the class based version. Looks like it's only used there, so you're right, it could be replaced with `showAdditionalButtons `.